### PR TITLE
feat: full analysis when int or float, max 1000 lines for other formats

### PR DESCRIPTION
**Pull request by geoffreyaldebert**

### CODE REVIEW

File: csv_detective/utils.py [hunk 1]
```diff
@@ -58,19 +59,26 @@ def test_col(table, all_tests, num_rows, output_mode):
...
         test_funcs[name] = {
             'func': test._is,
             'prop': test.PROPORTION
         }
-
     return_table = pd.DataFrame(columns=table.columns)
     for key, value in test_funcs.items():
+        # When analysis of all file is requested (num_rows = -1) we fix a threshold of 
+        # 1000 rows for every checks outside int or float format
+        local_num_rows = max(num_rows, 1000)
```
Comment (Pierlou) [inline csv_detective/utils.py:70]: If user inputs a value between 1 and 1000 for a column that is not in the list below, doesn't this force a larger analysis than requested?

File: csv_detective/utils.py [hunk 2]
```diff
@@ -58,19 +59,26 @@ def test_col(table, all_tests, num_rows, output_mode):
...
+        # For checks detecting int or float format, we analyze the whole file (because
+        # error can be generated afterward when exploiting this data into a database)
+        if key in ["int", "float", "longitude", "longitude_l93", "longitude_wgs", 
+            "longitude_wgs_fr_metropole", "latitude", "latitude_l93", "latitude_wgs", "latitude_wgs_fr_metropole", "iso_country_code_numeric"
+        ]:
+            local_num_rows = max(-1, num_rows)
```
Comment (Pierlou) [inline csv_detective/utils.py:76]: How about (for lines 70 to 76):
```
if num_rows==-1:

    if key in ["int", "float", "longitude", "longitude_l93", "longitude_wgs", 
            "longitude_wgs_fr_metropole", "latitude", "latitude_l93", "latitude_wgs", "latitude_wgs_fr_metropole", "iso_country_code_numeric"
        ]:

        local_num_rows=-1

    else:

        local_num_rows=1000

else:

    local_num_rows=min(rows,1000)
```

File: csv_detective/utils.py (hunk 1 above)
Comment (geoffreyaldebert) [reply csv_detective/utils.py:70]: hum, I don't think so. If num_rows = 500 for instance, every column will be analyzed for first 500 rows except for columns below where whole column will be analyzed. If num_rows > 1000 we will analyzed only first 1000 rows except for int or float checks

File: csv_detective/utils.py (hunk 2 above)
Comment (geoffreyaldebert) [reply csv_detective/utils.py:76]: With this : 
- If we setup an analyze to 500 rows, we will analyze more than user asked
- If we setup an analyze to 2000 rows, we will analyze less than user asked

File: csv_detective/utils.py (hunk 1 above)
Comment (geoffreyaldebert) [reply csv_detective/utils.py:70]: ok my bad I understand I will correct it

**Review by Pierlou — APPROVED** [review]

**Pull request merged**